src/Microsoft.AspNetCore.Mvc.Core/Controllers/DefaultControllerTypeProvider.cs

```diff
@@ -44,79 +44,48 @@ public virtual IEnumerable<TypeInfo> ControllerTypes
         /// Returns <c>true</c> if the <paramref name="typeInfo"/> is a controller. Otherwise <c>false</c>.
         /// </summary>
         /// <param name="typeInfo">The <see cref="TypeInfo"/>.</param>
-        /// <param name="candidateAssemblies">The set of candidate assemblies.</param>
         /// <returns><c>true</c> if the <paramref name="typeInfo"/> is a controller. Otherwise <c>false</c>.</returns>
-        protected internal virtual bool IsController(
-            TypeInfo typeInfo,
-            ISet<Assembly> candidateAssemblies)
+        protected internal virtual bool IsController(TypeInfo typeInfo)
         {
             if (typeInfo == null)
             {
                 throw new ArgumentNullException(nameof(typeInfo));
             }
 
-            if (candidateAssemblies == null)
-            {
-                throw new ArgumentNullException(nameof(candidateAssemblies));
-            }
-
             if (!typeInfo.IsClass)
             {
                 return false;
             }
+
             if (typeInfo.IsAbstract)
             {
                 return false;
             }
+
             // We only consider public top-level classes as controllers. IsPublic returns false for nested
             // classes, regardless of visibility modifiers
             if (!typeInfo.IsPublic)
             {
                 return false;
             }
+
             if (typeInfo.ContainsGenericParameters)
             {
                 return false;
             }
-            if (!typeInfo.Name.EndsWith(ControllerTypeName, StringComparison.OrdinalIgnoreCase) &&
-                !DerivesFromController(typeInfo, candidateAssemblies))
-            {
-                return false;
-            }
+
             if (typeInfo.IsDefined(typeof(NonControllerAttribute)))
```

Comment:
I don't think so, you still want to have the ability to explictly opt out of being a controller.